### PR TITLE
Modernize the setup infractructure slightly.

### LINKED ISSUE #96: `python -mbuild` does not work
Modern Python building does not work:
```
[snip]
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools >= 40.8.0, wheel)
* Getting build dependencies for wheel...
fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
  File "/home/matwolf/.local/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/home/matwolf/.local/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/home/matwolf/.local/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
    self.run_setup()
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/build_meta.py", line 480, in run_setup
    super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/build_meta.py", line 311, in run_setup
    exec(code, locals())
  File "<string>", line 40, in <module>
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/__init__.py", line 103, in setup
    return distutils.core.setup(**attrs)
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 147, in setup
    _setup_distribution = dist = klass(attrs)
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/dist.py", line 314, in __init__
    self.metadata.version = self._normalize_version(self.metadata.version)
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/dist.py", line 350, in _normalize_version
    normalized = str(Version(version))
  File "/tmp/build-env-et6jllza/lib/python3.8/site-packages/setuptools/_vendor/packaging/version.py", line 198, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
setuptools.extern.packaging.version.InvalidVersion: Invalid version: ''

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
```

### PR DESCRIPTION
**Pull request by matz-e**
Closes #96 by removing `git describe …` in favor of `setuptools_scm`.
This will also allow to build fully functional sdist files in the
future.

Also relax the scipy requirement for testing with modern Python
containers.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -5,7 +5,6 @@ name = neurodamus
 envlist = flake8, unit
 
 [testenv]
-package = wheel
```
[the remark this answers is not in the document]
Comment (ferdonline) [reply tox.ini:8]: What is the reason to go back to sdist?

File: setup.py [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-    v = subprocess.run(['git', 'describe', '--tags'],
-                       stdout=subprocess.PIPE).stdout.strip().decode()
-    __version__ = v[: v.rfind("-")].replace("-", ".dev") if "-" in v else v
-except Exception as e:
-    raise RuntimeError("Could not get version from Git repo") from e
 
-
-package_info = dict(
+setup(
     name='neurodamus',
```
Comment (ferdonline) [inline setup.py:5]: This destroys the ability of setup.py being importable. It it sometimes useful for getting information

**Comment by ferdonline** [comment]
Thanks for the patch. Just some nitpicks.
Btw, do you think we can avoid this somewhat annoying scm behavior "i.e. it automatically adds all of the SCM-managed files to the sdist"?

**Comment by matz-e** [comment]
> Btw, do you think we can avoid this somewhat annoying scm behavior "i.e. it automatically adds all of the SCM-managed files to the sdist"?

What do you mean? It adds some versioning info to the `sdist` so that it can be built automatically without a full `git` history. All version info is pre-compiled into `PKG-INFO` etc

**Comment by ferdonline** [comment]
> > Btw, do you think we can avoid this somewhat annoying scm behavior "i.e. it automatically adds all of the SCM-managed files to the sdist"?
> 
> What do you mean? It adds some versioning info to the `sdist` so that it can be built automatically without a full `git` history. All version info is pre-compiled into `PKG-INFO` etc. Other stuff should be managed via the manifest.

I mean what's written in bold in https://github.com/pypa/setuptools_scm#about

**Comment by ferdonline** [comment]
> Ah. I pushed a MANIFEST that excludes documentation and tests. The `sdist` is then only 450 kB.

Ah nice! I was misled by that #about, that it would include `.git`

**Review by ferdonline — APPROVED** [review]

**Comment by ferdonline** [comment]
Thanks! Dropping since there's #98

**Pull request closed without merging**